src/node_mime.cc

```diff
@@ -0,0 +1,369 @@
+              } else {
+                right += 1;
+                if (right < src.size()) {
+                  c = src[right];
+                  if ((c & 0xC0) != 0x80) {
+                    is_invalid = true;
+                  }
+                } else {
+                  is_invalid = true;
+                }
+              }
+            } else if (is_3_byte) {
+              is_invalid = true;
+              right += 2;
+            } else if (is_4_byte) {
+              is_invalid = true;
+              right += 3;
+            } else if (!is_http_quoted_string_token(c)) {
+              is_invalid = true;
+            }
+            if (is_invalid && !saw_invalid_char) {
+              saw_invalid_value = true;
+              first_invalid_char = right;
+            }
+          }
+        }
+        {
+          auto trim_right = right;
+          while (is_ascii_whitespace(src[trim_right - 1])) {
+            trim_right--;
+          }
+          if (!saw_invalid_char || first_invalid_char >= trim_right) {
+            parameterValue = src.substr(left, trim_right - left);
+          }
+        }
+      }
+    }
+    if (parameterName.size() > 0 &&
+        parameterValue.size() > 0 &&
+        !saw_invalid_value) {
+      parameters.push_back(std::pair<std::string, std::string>(
+          parameterName,
+          parameterValue));
+    }
+  }
+}
+
+
+void MIMEParser::Parse(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  auto context = env->context();
+
+  Isolate* isolate = args.GetIsolate();
+  HandleScope scope(isolate);
+
+  if (args.IsConstructCall()) {
+    env->ThrowError("parse() must not be called using new");
+    return;
+  }
+
+  if (!args[0]->IsString()) {
+    env->ThrowError("first argument is not a string");
+    return;
+  }
+  if (!args[1]->IsObject()) {
+    env->ThrowError("second argument is not an object");
+    return;
+  }
+
+  auto that = args[1]->ToObject();
+
+  Local<String> string = args[0].As<String>();
+  v8::String::Utf8Value source(isolate, string);
+
+  MIME mime(std::string(*source, source.length()));
+
+  if (mime.flags_ != mime_flags::MIME_FLAGS_NONE) {
+    env->ThrowError("Error parsing MIME");
+    return;
+  }
+
+  auto type_key = FIXED_ONE_BYTE_STRING(isolate, "type");
+  auto subtype_key = FIXED_ONE_BYTE_STRING(isolate, "subtype");
+  auto parameters_key = FIXED_ONE_BYTE_STRING(isolate, "parameters");
+  Local<String> type =
+      String::NewFromUtf8(isolate,
+                          mime.type_.c_str(),
+                          v8::NewStringType::kInternalized,
+                          mime.type_.length()).ToLocalChecked();
+  Local<String> subtype =
+      String::NewFromUtf8(isolate,
+                          mime.subtype_.c_str(),
+                          v8::NewStringType::kInternalized,
+                          mime.subtype_.length()).ToLocalChecked();
+  that->Set(type_key, type);
+  that->Set(subtype_key, subtype);
```

Comment:
Ditto for these overloads, they are deprecated and should use a `context` argument, like: `that->Set(context, type_key, type).FromJust()`

Reply:
fixed